packages/material-ui-lab/src/MasonryItem/MasonryItem.js

```diff
@@ -0,0 +1,161 @@
+import MasonryContext from '../Masonry/MasonryContext';
+
+const useUtilityClasses = (styleProps) => {
+  const { classes } = styleProps;
+
+  const slots = {
+    root: ['root'],
+  };
+
+  return composeClasses(slots, getMasonryItemUtilityClass, classes);
+};
+
+export const style = ({ styleProps, theme }) => {
+  let styles = {
+    width: '100%',
+    '& > *': {
+      // all contents should have a width of 100%
+      width: '100%',
+      boxSizing: 'inherit',
+      ...(styleProps.isSSR && { height: '100%' }),
+    },
+    visibility: styleProps.height ? 'visible' : 'hidden',
+    gridColumnEnd: `span ${styleProps.columnSpan}`,
+    boxSizing: 'inherit',
+  };
+
+  const base = {};
+  Object.keys(theme.breakpoints.values).forEach((breakpoint) => {
+    if (styleProps.spacing[breakpoint] != null) {
+      base[breakpoint] = true;
+    }
+  });
+  const spacingValues = resolveBreakpointValues({ values: styleProps.spacing, base });
+  const transformer = createUnarySpacing(theme);
+  const styleFromPropValue = (propValue) => {
+    const gap = styleProps.height ? Number(getValue(transformer, propValue).replace('px', '')) : 0;
+    // For lazy-loaded images to load properly, masonry item should take up space greater than 1px.
+    // Taking into account a row gap of 2px, rowSpan should at least be 2.
+    const rowSpan = styleProps.height ? Math.ceil((styleProps.height + gap) / 2) : 2;
+    return {
+      gridRowEnd: `span ${rowSpan}`,
+      paddingBottom: gap - 2,
+    };
+  };
+  styles = { ...styles, ...handleBreakpoints({ theme }, spacingValues, styleFromPropValue) };
+
+  return styles;
+};
+
+const MasonryItemRoot = styled('div', {
+  name: 'MuiMasonryItem',
+  slot: 'Root',
+  overridesResolver: (props, styles) => {
+    return [styles.root];
+  },
+})(style);
+
+const MasonryItem = React.forwardRef(function MasonryItem(inProps, ref) {
+  const props = useThemeProps({
+    props: inProps,
+    name: 'MuiMasonryItem',
+  });
+
+  const masonryItemRef = React.useRef(null);
+
+  const { spacing = 1 } = React.useContext(MasonryContext);
+  const { children, className, component = 'div', columnSpan = 1, defaultHeight, ...other } = props;
+  const isSSR = defaultHeight !== undefined;
+
+  const [height, setHeight] = React.useState(defaultHeight);
+
+  const styleProps = {
+    ...props,
+    isSSR,
+    spacing,
+    columnSpan,
+    height: height < 0 ? 0 : height, // MasonryItems to which negative or zero height is passed will be hidden
+  };
+
+  const classes = useUtilityClasses(styleProps);
+
+  React.useEffect(() => {
+    // do not create a resize observer in case of SSR masonry
+    if (isSSR) {
+      return () => {};
+    }
+    const resizeObserver = new ResizeObserver(([item]) => {
+      setHeight(item.contentRect.height);
+    });
+    const item = masonryItemRef.current.firstChild;
+    resizeObserver.observe(item);
+    return () => {
+      resizeObserver.unobserve(item);
+    };
+  }, [isSSR, height]);
```

Comment:
```suggestion
  }, [isSSR]);
```

`height` as a dependency is now obsolete.

Reply:
Gottcha. Thank you!